### ISSUE TITLE
Logpush null filter results in unexpected end of JSON input

### ISSUE THREAD

**Issue by sawp-d** [cloudflare/terraform-provider-cloudflare#1703]
### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

terraform: 1.1.7
cloudflare: 3.17.0 (was using 3.16.0 to create the existing logpush job)

### Affected resource(s)

resource cloudflare_logpush_job

### Terraform configuration files

```hcl
resource "cloudflare_logpush_job" "example_job" {
  enabled             = true
  zone_id             = "..."
  name                = "..."
  logpull_options     = "fields=..."
  destination_conf    = "..."
  ownership_challenge = "..."
  dataset             = "..."
}
```

### Debug output

Can provide if necessary - started to run but way too many lines of logs that it broke build :( 

### Expected output

Expected filter = optional, logpush job creation successful

### Actual output

For apply/destroy: 
```
Error: error parsing logpush job from resource: unexpected end of JSON input
```

### Steps to reproduce

1. Create new logpush job without filter attribute -> will fail? Haven't tested yet
2. Update existing logpush job -> will fail, error message, adds filter = null to state
3. Try destroying existing logpush job -> will fail, error message

**Comment by cmackenzie1** [comment]
fyi: @nuggien

**Comment by jacobbednarz** [comment]
closed by cloudflare/cloudflare-go#937
